src/tree/v2/providers/CompatibleBranchDataProvider.ts

```diff
@@ -0,0 +1,69 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtParentTreeItem, AzExtTreeDataProvider, AzExtTreeItem, IFindTreeItemContext, ITreeItemPickerContext } from '@microsoft/vscode-azext-utils';
+import { AppResourceResolver } from '@microsoft/vscode-azext-utils/hostapi';
+import * as vscode from 'vscode';
+import { ApplicationResource, BranchDataProvider, ResourceModelBase } from '../../../api/v2/v2AzureResourcesApi';
+
+export class CompatibleBranchDataProvider<TResource extends ApplicationResource, TModel extends AzExtTreeItem & ResourceModelBase> extends AzExtTreeDataProvider implements BranchDataProvider<TResource, TModel> {
+    private readonly overrideOnDidChangeTreeDataEmitter = new vscode.EventEmitter<TModel | undefined>();
+
+    public constructor(private readonly resolver: AppResourceResolver, loadMoreCommandId: string) {
+        super(undefined as unknown as AzExtParentTreeItem, loadMoreCommandId);
+    }
+
+    //#region TreeDataProvider
+
+    public override readonly onDidChangeTreeData = this.overrideOnDidChangeTreeDataEmitter.event;
+
+    public override getParent(treeItem: TModel): Promise<TModel> {
+        // This method is redeclared to make TypeScript happier, but it's no more than a super call with extra casts
+        return super.getParent(treeItem) as Promise<TModel>;

```

Comment:
This is operating on the assumption that if the parent of the AppResource node itself is requested, that RG will trap and answer that--because it _is_ technically the owner of that node.